x/confidentialtransfers/keeper/msg_server.go

```diff
@@ -2,48 +2,450 @@ package keeper
+	address, err := sdk.AccAddressFromBech32(instruction.FromAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Invalid Address")
+	}
+
+	_, exists := m.Keeper.GetAccount(ctx, address, instruction.Denom)
+	if exists {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Account already exists")
+	}
 
-func (m msgServer) Deposit(ctx context.Context, deposit *types.MsgDeposit) (*types.MsgDepositResponse, error) {
-	//TODO implement me
-	panic("implement me")
+	// Validate the public key
+	validated := zkproofs.VerifyPubKeyValidity(*instruction.Pubkey, *instruction.Proofs.PubkeyValidityProof)
+	if !validated {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid public key")
+	}
+
+	// Validate the pending balance lo is zero.
+	validated = zkproofs.VerifyZeroBalance(instruction.Proofs.ZeroPendingBalanceLoProof, instruction.Pubkey, instruction.PendingBalanceLo)
+	if !validated {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid pending balance lo")
+	}
+
+	// Validate the pending balance hi is zero.
+	validated = zkproofs.VerifyZeroBalance(instruction.Proofs.ZeroPendingBalanceHiProof, instruction.Pubkey, instruction.PendingBalanceHi)
+	if !validated {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid pending balance hi")
+	}
+
+	// Validate the available balance is zero.
+	validated = zkproofs.VerifyZeroBalance(instruction.Proofs.ZeroAvailableBalanceProof, instruction.Pubkey, instruction.AvailableBalance)
+	if !validated {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid available balance")
+	}
+
+	// Create the account
+	account := types.Account{
+		PublicKey:                   *instruction.Pubkey,
+		PendingBalanceLo:            instruction.PendingBalanceLo,
+		PendingBalanceHi:            instruction.PendingBalanceHi,
+		AvailableBalance:            instruction.AvailableBalance,
+		DecryptableAvailableBalance: instruction.DecryptableBalance,
+		PendingBalanceCreditCounter: 0,
+	}
+
+	// Store the account
+	m.Keeper.SetAccount(ctx, address, req.Denom, account)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeInitializeAccount,
+			sdk.NewAttribute(types.AttributeDenom, instruction.Denom),
+			sdk.NewAttribute(types.AttributeAddress, instruction.FromAddress),
+		),
+	})
+
+	return &types.MsgInitializeAccountResponse{}, nil
 }
 
-func (m msgServer) Withdraw(ctx context.Context, withdraw *types.MsgWithdraw) (*types.MsgWithdrawResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (m msgServer) Deposit(goCtx context.Context, req *types.MsgDeposit) (*types.MsgDepositResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Validate request
+	err := req.ValidateBasic()
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Invalid Request")
+	}
+
+	// Check if the account exists
+	address, err := sdk.AccAddressFromBech32(req.FromAddress)
```

Comment:
Refactored the keeper methods (get, set and delete accounts) to take in address as a string instead and try to perform this conversion, since the only place that needs this sdkAccAddress is the keeper for now.